linux_os/guide/system/software/updating/security_patches_up_to_date/rule.yml

```diff
@@ -60,21 +60,19 @@ references:
     cis-csc: 18,20,4
     anssi: BP28(R08)
 
-{{# Make sure all the external OVAL content links are secured via TLS! #}}
 
 # SCAP 1.3 content should reference flat non compressed xml files
-{{% if product == "rhel7" %}}
-oval_external_content: "https://www.redhat.com/security/data/oval/com.redhat.rhsa-RHEL7.xml"
-{{% elif product == "rhel8" %}}
-oval_external_content: "https://www.redhat.com/security/data/oval/com.redhat.rhsa-RHEL8.xml"
-{{% elif product == "ubuntu1604" %}}
-oval_external_content: "https://people.canonical.com/~ubuntu-security/oval/com.ubuntu.xenial.cve.oval.xml"
-{{% elif product in ["ol7", "ol8"] %}}
-oval_external_content: "https://linux.oracle.com/security/oval/com.oracle.elsa-all.xml.bz2"
-{{% elif product == "sle12" %}}
-oval_external_content: "https://support.novell.com/security/oval/suse.linux.enterprise.12.xml"
-{{% elif product == "sle15" %}}
-oval_external_content: "https://support.novell.com/security/oval/suse.linux.enterprise.15.xml"
+{{% if oval_feed_url %}}
+oval_external_content: {{{ oval_feed_url }}}
+  {{% if not oval_feed_url.startswith("https") %}}
```

Comment:
This needs to fail in python code rather than through the jinja template.

Reply:
Where specifically would you suggest to handle the issue? What's wrong with having it here, where other aspects of the feed URL are translated to the rule's XCCDF?

Reply:
What if the `oval_feed_url` was process in [`open_environment`](https://github.com/ComplianceAsCode/content/blob/master/ssg/yaml.py#L141) ?

That is the function that is, in general, loading `product.yml`.

Reply:
> What if the oval_feed_url was processed in open_environment ?

Technically, it could go there, but how would it be better than the current approach which works? Right now, we don't make any assumptions about the schema of `product.yml` - we just propagate its contents to the environment without asking questions. 

Acknowledging that the product can have e.g. the `oval_feed_url` property that, for some reason, shouldn't start with `http` means that the Python code becomes responsible for validation of `product.yml`. As we don't do any validation of products now, it is likely that starting to do so for the last property that caught attention would be inconsistent with how other properties are handled.

Reply:
> Technically, it could go there, but how would it be better than the current approach which works?

The Rule becomes more readable, and the build would still fail if the feed url is not secure.

> Acknowledging that the product can have e.g. the `oval_feed_url` property that, for some reason, shouldn't start with `http` means that the Python code becomes responsible for validation of `product.yml`. As we don't do any validation of products now, it is likely that starting to do so for the last property that caught attention would be inconsistent with how other properties are handled.

Not validating every property may be inconsistent, but it should be ok to only validate the properties that need validation.

Reply:
Yes. The readability of this rule is now quite atrocious due to these changes.

Reply:
OK, thank you for the clarification.